_posts/ericabrody/2016-04-29-ericarbrody-Final-Project.md

```diff
@@ -0,0 +1,49 @@
+
+Having such a methodical approach to the programming was really helpful. It allowed me to break the coding process down into really small bits that I could time box effectively. Also, this process helped me feel more comfortable with the programming tasks. I spent a lot of time getting the first three functions to work, with a lot of breaks to clear my head, but these served as a good foundation for the rest of the work. When I would get very frustrated with a problem or just not know how to start, I would search Python documentation on the internet to try to figure it out. For example, I googled “Python sorting” and google had an auto-complete option “Python sorting dictionary by value,” which led me to this page (https://docs.python.org/3/howto/sorting.html). I had to read the documentation a few different times and I ended up using some code that used “lambda” and I wasn’t totally sure how lambda worked, but it helped me print my frequency table the way I wanted with the response values in order of response value, rather than the highest response value first. Once, when I got totally frustrated trying to format the output for the univariate statistics of the continuous variables, Eric recommended that I look up Python format strings and I found some helpful resources – https://pyformat.info and http://www.python-course.eu/python3_formatted_output.php.  
+
+I also had to do a lot of trial and error. Every time I got some code together to run, I would get at least one error – a missing colon or something and often bigger errors. I appreciated that Cloud 9 is fast, so I could keep rerunning the programs with small changes. Throughout the process, my ability to figure out what the errors meant or at least how to troubleshoot got better. I printed a lot of ugly output to find mistakes.    
+
+I think the most valuable problem-solving approach I used was to walk away from the project when I got stuck. Surprisingly, I had to walk away from both the computer and my paper and pencil work. In lieu of the E. Hauser shower approach, I took a lot of little naps. Or tried to, once I laid down and closed my eyes, my brain would keep working on the problem and frequently I would get up after about 10 minutes with an idea of how to fix the code. Even if that initial idea didn’t work, it would get me one step closer to the solution.
```

Comment:
Naps are awesome!  They help your background brain do heavy lifting.  Check this classic talk by Rich Hickey: https://www.youtube.com/watch?v=f84n5oFoZBc